api/nakadi-event-bus-api.yaml

```diff
@@ -623,6 +539,349 @@ paths:
+              items:
+                description: list of subscriptions
+                type: array
+                items:
+                  $ref: '#/definitions/Subscription'
+            required:
+              - items
+              - _links
+
+  /subscriptions/{subscription_id}:
+    get:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Returns a subscription identified by id.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/Subscription'
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+
+  /subscriptions/{subscription_id}/cursors:
+    get:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Exposes the currently committed offsets of a subscription.
+      parameters:
+        - $ref: '#/parameters/SubscriptionId'
+      responses:
+        '200':
+          description: Ok
+          schema:
+            type: object
+            properties:
+              items:
+                description: list of cursors for subscription
+                type: array
+                items:
+                  $ref: '#/definitions/Cursor'
+            required:
+              - items
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+    post:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        Endpoint for committing offsets of the subscription. The client must commit at least once
+        every 60 seconds, otherwise Nakadi will consider the client to be gone and will close the
+        connection.
+
+        When a batch is committed that also automatically commits all previous batches that were
+        sent in a stream for this partition.
+      parameters:
+        - name: subscription_id
+          in: path
+          type: string
+          description: Id of subscription
+          required: true
+        - name: X-Nakadi-StreamId
+          in: header
+          type: string
+          description: |
+            Id of stream which client uses to read events. It is not possible to make a commit for a terminated or
+            none-existing stream. Also the client can't commit something which was not sent to his stream.
+        - name: cursors
+          in: body
+          schema:
+            type: object
+            properties:
+              cursors:
+                description: |
+                  List of cursors that the consumer acknowledges to have successfully processed.
+                type: array
+                items:
+                  $ref: '#/definitions/Cursor'
+            required:
```

Comment:
> As an author of libraries I'll use a generic data structure for cursors (JsonNode) since that would make me resilient against changes like that.

Right, but why do that when a definition can be created by design?

To ground this a bit, here's a SubscriptionCursor with an `event_type_name` field:

```yaml
   SubscriptionCursor:
      required:
        - id
        - partition
        - offset
        - event_type_name
        - cursor_token
     properties:
       id:
         type: string
         format: uuid
         description: |
           Id of the cursor (a UUID).
       partition:
         type: string
         description: |
           Id of the partition pointed to by this cursor.
       offset:
         type: string
          description: |
            Offset of the event being pointed to.
       event_type_name:
         type: string
          description: |
            The name of the event type this partition's events belong to
       cursor_token:
         type: string
          description: |
            An opaque value defined by the server.
```

The other additions are `id` and `cursor_token`. You could quibble about `id` being needed (valuable for logging and diagnostics, not strictly needed for checkpointing to work). The `cursor_token` is future-proofing; basically it's a cookie placeholder to allow opaque state to be embedded by the server and roundtripped back by the client in the event something more than `event_type_name` is needed. All non-optional which means clients can't ignore them.